### ISSUE TITLE
Rename BaseFee stuff to avoid confusion.

### ISSUE THREAD

**Issue by SpicyLemon** [provenance-io/provenance#1096]
## Summary

Rename the functions and variables named "Base Fee" to avoid confusion due to overloading it's meaning.

## Problem Definition

We have an event attribute key of `baseFee` that contains the value `fee - additional fees`. However, there are several functions and variables in our code referencing `baseFee` that use it to mean other things (e.g. `floor gas price * gas`).

## Proposal

Because the event attributes are outward facing, and changing them is annoying for consumers, we should **NOT** change them. Similarly, we cannot change the `minimum-gas-prices` config field. Instead we should rename the functions and variables in our code to either a) more closely resemble those attribute/field names or b) at least not conflict with the meaning of the attributes/fields.

Here are some definitions I'd like to standardize on:
* (unmodified) "fee" or `feeCoins` = the full fee that the user provided as part of the `tx`.
* "additional fee" = total of the fees that are required by the `x/msgfees` module as a result of the msg type that's being processed (including the `MsgAssessCustomMsgFeeRequest` stuff).
* "base fee" = fee - additional fees = floor fee + overage = the value of the `baseFee` event attribute.
* "min fee" or "floor fee" = floor gas price (controlled by the `x/msgfees` module params) * gas wanted. In my opinion, it would be better to refer to this as the "floor fee" as much as possible to avoid confusion with the "min gas prices". However, then it's not named in line with the `min_fee_charged` event attribute that it ends up in. So some discretion will be needed here.
* "min gas prices" references the `minimum-gas-prices` configuration field. Unfortunately, it will have to clash with the `min_fee_charged` until we revamp/clean-up the events (and can hopefully rename `min_fee_charged`). Luckily, we only really care about it in one place: the `MinGasPricesDecorator` where we don't care about the floor fee.

An example of something to fix: there's a function named 'CalculateBaseFee` that actually calculates what ends up being the `min_fee_charged` aka "floor fee". It's confusing because it does NOT calculate what ends up being the "baseFee" in the events.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by nullpointer0x00** [comment]
I agree with the need to change the names of these things to be more consistent and clear to what is going on.  It will save time in the long run for implementing any changes in the future.

**Comment by iramiller** [comment]
I think we can discuss further with @minxylynx about how the explorer is using these existing values ... but making a logical standard and documenting it is very important here.  The `min-gas-prices` attribute and how it impacts the network (only as a node filter) is not something that is well understood ... when you mix that concept in with the rest of our enhancements things are likely to come further off the rails... 

The 'additional fees' being `msg fees` is kind of oddly named when taken into account with everything else as well... these would potentially not be obvious as standard msg fees in a network that did not have base gas fees for example ... but I agree that we probably have a bit less flexibility in changing this value in the published events.
